src/report/example.rs

```diff
@@ -0,0 +1,82 @@
+use std::convert::From;
+
+#[cfg(feature = "expectest_compat")]
+use expectest::core::TestResult as ExpectestResult;
+
+#[derive(Clone, PartialEq, Eq, Debug)]
+pub struct Failure {
+    pub message: Option<String>,
+}
+
+impl Failure {
+    pub fn new<S>(message: Option<S>) -> Self
+    where
+        String: From<S>,
+    {
+        Failure { message: message.map(|s| s.into()) }
+    }
+}
+
+#[derive(Clone, PartialEq, Eq, Debug)]
+pub enum ExampleReport {
+    Success,
+    Ignored,
+    Failure(Failure),
+}
+
+impl ExampleReport {
+    pub fn err<S>(message: Option<S>) -> Self
+    where
+        String: From<S>,
+    {
+        ExampleReport::Failure(Failure::new(message))
+    }
+
+    pub fn is_ok(&self) -> bool {
+        *self == ExampleReport::Success
```

Comment:
I would prefer `self == &ExampleReport::Success` as it feels more neutral.

With an `*` it's like it's behind a `Box` or a `Mutex`